pkg/apis/cluster/v1alpha1/machine_types.go

```diff
@@ -132,6 +132,10 @@ type MachineStatus struct {
 	// serialized/deserialized from this field.
 	// +optional
 	ProviderStatus *runtime.RawExtension `json:"providerStatus,omitempty"`
+        
+	// ProviderID is the identification ID of the machine provided by the provider
+	// +optional
+	ProviderID *string `json:"providerID,omitempty"`
```

Comment:
Please update this comment to make it clear that this field must match the provider ID as seen on the node object corresponding to this machine.